core/trino-spi/src/main/java/io/trino/spi/block/Int96ArrayBlock.java

```diff
@@ -250,13 +249,35 @@ public String toString()
         return sb.toString();
     }
 
-    Slice getHighSlice()
+    public static void encodeInt96(long high, int low, int[] values, int position)
+    {
+        int offset = position * 3;
+        encodeHigh(high, values, position);
+        values[offset + 2] = low;
+    }
+
+    static void encodeHigh(long high, int[] values, int position)
+    {
+        values[(position * 3)] = (int) high;
+        values[(position * 3) + 1] = (int) (high >>> 32);
```

Comment:
This results in a weird layout for the 96-bit value.

Given a 96-bit value of the shape `HHHHMMMMLLLL`, the encoding as ints will be `MMMM HHHH LLLL`

We should either do big-endian and encode it as `HHHH MMMM LLLL` or little-endian, and encode it as `LLLL MMMM HHHH` (with each of the integers encoded as big-endian or little-endian respectively)

Reply:
After talking, I renamed Int96 to Fixed12, and removed the terms "high" and "low" to avoid any confusion with traditional numeric encoding.